test/Test/Spec/RunnerSpec.purs

```diff
@@ -1,35 +1,58 @@
+import Test.Spec (Item(..), Spec, SpecT(..), Tree(..), describe, it)
 import Test.Spec.Assertions (shouldEqual)
 import Test.Spec.Fixtures (itOnlyTest, describeOnlyNestedTest, describeOnlyTest, sharedDescribeTest, successTest)
-import Test.Spec.Runner (runSpec)
+import Test.Spec.Tree (discardUnfocused)
 
 runnerSpec :: Spec Unit
 runnerSpec =
   describe "Test" $
     describe "Spec" $
       describe "Runner" do
         it "collects \"it\" and \"pending\" in Describe groups" do
-          results <- runSpec successTest
-          results `shouldEqual` [Describe false "a" [Describe false "b" [It false "works" Success]]]
+          runSpecFocused successTest `shouldEqual`
+            [ Node (Left "a")
+              [ Node (Left "b") [ Leaf "works" $ Just false ]
+              ]
+            ]
         it "collects \"it\" and \"pending\" with shared Describes" do
-          results <- runSpec sharedDescribeTest
-          results `shouldEqual` [Describe false "a" [Describe false "b" [It false "works" Success],
-                                                     Describe false "c" [It false "also works" Success]]]
+          runSpecFocused sharedDescribeTest `shouldEqual`
+            [ Node (Left "a")
+              [ Node (Left "b") [ Leaf "works" $ Just false ]
+              , Node (Left "c") [ Leaf "also works" $ Just false ]
+              ]
+            ]
         it "filters using \"only\" modifier on \"describe\" block" do
-          results <- runSpec describeOnlyTest
-          results `shouldEqual` [Describe true "a" [Describe false "b" [It false "works" Success],
-                                                    Describe false "c" [It false "also works" Success]]]
+          runSpecFocused describeOnlyTest `shouldEqual`
+            [ Node (Left "a")
+              [ Node (Left "b") [ Leaf "works" $ Just true ]
+              , Node (Left "c") [ Leaf "also works" $ Just true ]
+              ]
+            ]
         it "filters using \"only\" modifier on nested \"describe\" block" do
-          results <- runSpec describeOnlyNestedTest
-          results `shouldEqual` [Describe true "b" [It false "works" Success]]
+          runSpecFocused describeOnlyNestedTest `shouldEqual`
+            [ Node (Left "a")
+              [ Node (Left "b") [ Leaf "works" $ Just true ]
+              ]
+            ]
         it "filters using \"only\" modifier on \"it\" block" do
-          results <- runSpec itOnlyTest
-          results `shouldEqual` [It true "works" Success]
+          runSpecFocused itOnlyTest `shouldEqual`
+            [ Node (Left "a")
+              [ Node (Left "b") [ Leaf "works" $ Just true ]
+              ]
+            ]
         it "supports async" do
           res <- delay (Milliseconds 10.0) *> pure 1
           res `shouldEqual` 1
+  where
+    runSpecFocused t = discardUnfocused (execWriter $ un SpecT t) <#> bimap (const unit) (un Item >>> _.isFocused)
```

Comment:
`runSpecFocused`, if I'm not misunderstanding, isn't running the full `runSpec` machinery, like this test did before. It seems it's collecting a tree structure of `Maybe Bool` values for `isFocused`. Would it be possible to rewrite this to use runSpec (or some underlying function) to exercise the runner module?

Reply:
Most I could do is to add `collect` let me know if there is something you think is worth adding/changing.